src/gt4py/next/otf/binding/nanobind.py

```diff
@@ -187,7 +187,47 @@ def make_argument(name: str, type_: ts.TypeSpec) -> str | BufferSID | CompositeS
     elif isinstance(type_, ts.ScalarType):
         return name
     else:
-        raise ValueError(f"Type '{type_}' is not supported in pybind11 interfaces.")
+        raise ValueError(f"Type '{type_}' is not supported in nanobind interfaces.")
+
+
+nanobind_sid_conv = """\
+#include <nanobind/nanobind.h>
```

Comment:
I still would prefer to have this in gridtools, e.g. if we improve that adapter (e.g. by passing over compile time sizes != any) all potential users will profit... If you don't like to do the move now, I will probably do it eventually.